src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -210,29 +203,52 @@ protected Object fetchField(ExecutionContext executionContext, ExecutionStrategy
 
         InstrumentationFieldFetchParameters instrumentationFieldFetchParams = new InstrumentationFieldFetchParameters(executionContext, fieldDef, environment);
         InstrumentationContext<Object> fetchCtx = instrumentation.beginFieldFetch(instrumentationFieldFetchParams);
-        Object fetchedValue = null;
+        CompletableFuture<?> fetchedValue;
+        DataFetcher dataFetcher = fieldDef.getDataFetcher();
+        dataFetcher = instrumentation.instrumentDataFetcher(dataFetcher, instrumentationFieldFetchParams);
         try {
-            DataFetcher dataFetcher = fieldDef.getDataFetcher();
-            dataFetcher = instrumentation.instrumentDataFetcher(dataFetcher, instrumentationFieldFetchParams);
-            fetchedValue = dataFetcher.get(environment);
-
-            fetchCtx.onEnd(fetchedValue);
+            Object fetchedValueRaw = dataFetcher.get(environment);
+            if (fetchedValueRaw instanceof CompletableFuture) {
+                fetchedValue = (CompletableFuture<?>) fetchedValueRaw;
+            } else {
+                fetchedValue = CompletableFuture.completedFuture(fetchedValueRaw);
+            }
         } catch (Exception e) {
-            fetchCtx.onEnd(e);
-
-            DataFetcherExceptionHandlerParameters handlerParameters = DataFetcherExceptionHandlerParameters.newExceptionParameters()
-                    .executionContext(executionContext)
-                    .dataFetchingEnvironment(environment)
-                    .argumentValues(argumentValues)
-                    .field(field)
-                    .fieldDefinition(fieldDef)
-                    .path(parameters.path())
-                    .exception(e)
-                    .build();
-
-            dataFetcherExceptionHandler.accept(handlerParameters);
+            fetchedValue = new CompletableFuture<>();
+            fetchedValue.completeExceptionally(e);
         }
-        return fetchedValue;
```

Comment:
I think we need to fix Instrumentation up.  

https://github.com/graphql-java/graphql-java/issues/601

We would use .whenComplete((result,throwable) -> ....) to do this.

This does not have to go in this PR